pydis_site/apps/api/viewsets/bot/off_topic_channel_name.py

```diff
@@ -106,6 +110,31 @@ def list(self, request: HttpRequest) -> Response:
                     'random_items': ["Must be a positive integer."]
                 })
 
+            if 'mark_used' in request.query_params and request.query_params['mark_used']:
+                queryset = self.get_queryset().order_by('?').exclude(used=True)[:random_count]
+                self.get_queryset().filter(
+                    name__in=(query.name for query in queryset)
+                ).update(used=True)
+
+                # When client request more channel names than non-used names is available, start
+                # new round of names.
```

Comment:
```suggestion
                # When the client requests more channel names than are available, 
                # we reset all names to used=True and start a new round of names.
```

Reply:
I'm pretty sure that this should `used=False` because then name is again available.

Reply:
oh, yeah. my bad.